staging/src/k8s.io/component-helpers/storage/volume/pv_helpers.go

```diff
@@ -225,19 +222,17 @@ func FindMatchingVolume(
 		}
 
 		// check if PV's DeletionTimeStamp is set, if so, skip this volume.
-		if utilfeature.DefaultFeatureGate.Enabled(features.StorageObjectInUseProtection) {
-			if volume.ObjectMeta.DeletionTimestamp != nil {
-				continue
-			}
+		if enableStorageObjectInUseProtection && volume.ObjectMeta.DeletionTimestamp != nil {
+			continue
 		}
 
 		nodeAffinityValid := true
 		if node != nil {
 			// Scheduler path, check that the PV NodeAffinity
 			// is satisfied by the node
-			// volumeutil.CheckNodeAffinity is the most expensive call in this loop.
+			// CheckNodeAffinity is the most expensive call in this loop.
 			// We should check cheaper conditions first or consider optimizing this function.
-			err := volumeutil.CheckNodeAffinity(volume, node.Labels)
+			err := CheckNodeAffinity(volume, node.Labels)
```

Comment:
@liggitt if this function moves to `scheduling`, then this package would still depend on it.

I think the dependency is reasonable.